torch/_inductor/codegen/common.py

```diff
@@ -215,13 +215,15 @@ def ctx():
             for _ in range(-offset):
                 self._indent -= 1
                 self.writeline("}")
-            yield
-            for _ in range(-offset):
-                self.writeline("{")
-                self._indent += 1
-            for _ in range(offset):
-                self._indent -= 1
-                self.writeline("}")
+            try:
+                yield
+            finally:
+                for _ in range(-offset):
+                    self.writeline("{")
```

Comment:
Some of these don't really make sense. If there's an exception why would we keep doing codegen?

Reply:
Hmm... in my rush I was thinking that you might throw an exception during codegen on purpose and then catch it. But that should already be covered by a try/except I think, and otherwise we shouldn't expect it to work.

Reply:
yeah

Reply:
Actually I'm not totally sure on which ones of these I should undo 🤔 What about like, the changes to `set_current_node`?